tmtools/helpers.py

```diff
@@ -0,0 +1,91 @@
+"""Helper tools to work with TM-Align."""
+
+import numpy as np
+
+
+def transform_coordinates(original_coords, tmalign_result):
+    """
+    Transform coordinates based on TM-Align result.
+
+    This function creates a copy of the input coordinates and applies the rotation
+    matrix and translation vector from a TM-align result to all atoms in the copied
+    structure.
+
+    Parameters
+    ----------
+    original_coords : array_like
+        The original coordinates to transform.
+    tmalign_result : TM_result
+        An object containing the TM-align transformation data, with attributes:
+        - u : array_like
+            3x3 rotation matrix.
+        - t : array_like
+            Translation vector with 3 elements.
+
+    Returns
+    -------
+    np.array
+        A new array object with the TM-align transformation applied.
+    """
+    # Create a copy of the structure to transform
+    aligned_coords = original_coords.copy()
+
+    # Create rotation matrix and translation vector as numpy arrays
+    rotation_matrix = np.array(tmalign_result.u, dtype=float)
+    translation_vector = np.array(tmalign_result.t, dtype=float)
+
+    # Iterate through all coordinates applying transformation
+    aligned_coords = np.apply_along_axis(
```

Comment:
With broadcasting, this can be done directly through a matrix multiplication:
```python
aligned_coords = coords @ u.T + t
```
This has the added advantage of being a bit faster, although that won't be noticeable unless you have really big proteins or end up converting a lot of them.

This notebook has some more details: https://colab.research.google.com/drive/12PttoMOaS8cPjWLUbGcvoyx_7yor3xBS?usp=sharing

Reply:
Oh! I didn't know that those where equivalent. Will change it to that since its faster and, IMO, cleaner.